docs: add readme and regex example for git-refs datasource (#19368)
renovatebot/renovate: lib/modules/datasource/git-refs/readme.md

> This datasource returns a ref from a Git repository. The `depName` must be a FQDN. To fetch the latest digest of a ref instead of the named ref, specify the ref as the `currentValue` and match on the `currentDigest`. This datasource can be used in combination with [regex managers](https://docs.renovatebot.com/modules/manager/regex/) to keep dependencies up-to-date which are not specifically supported by Renovate.

Reviewer: ```suggestion
This datasource can be used in combination with [regex managers](https://docs.renovatebot.com/modules/manager/regex/) to keep dependencies up-to-date which are not specifically supported by Renovate.

This datasource returns a reference from a Git repository.
The `depName` must be a fully qualified domain name.
To fetch the latest digest of a reference instead of the named reference, specify the reference as the `currentValue` and match on the `currentDigest`.
```

Changes:

- Our docs use the "one sentence per line pattern"
- Move the introductory sentence to the top
- Expand the terms `FQDN`, `ref`/`refs` and `repo` fully

> **Usage Example**

Reviewer: ```suggestion
**Usage example**
```

Changes:

 - Reduce capitalization in the heading

> A real-world example for this specific datasource would be maintaining the HEAD digest of the `master` branch of an unversioned and untagged repo in a config file. This can be achieved by configuring a generic regex manager in `renovate.json` for files named `versions.ini`:

Reviewer: ```suggestion
Say you want to maintain the HEAD digest of the `main` branch of an unversioned and untagged repository in a config file.
You would configure a generic regex manager in `renovate.json` for files named `versions.ini`:
```

Changes:

- Replace `master` with `main`
- Use the word `you` to pull the reader in the text
- Write short sentences and paragraphs

```json
{
  "regexManagers": [
    {
      "fileMatch": ["^versions.ini$"],
      "matchStrings": ["GOOGLE_API_VERSION=(?<currentDigest>.*?)\\n"],

>       "currentValueTemplate": "master",

Reviewer: ```suggestion
      "currentValueTemplate": "main",
```

Changes:

- Replace `master` with `main` here as well

This datasource can be used in combination with [regex managers](https://docs.renovatebot.com/modules/manager/regex/) to keep dependencies up-to-date which are not specifically supported by Renovate.

This datasource returns a reference from a Git repository.
The `depName` (or `packageName` if in use) must be a fully qualified domain name.
To fetch the latest digest of a reference instead of the named reference, specify the reference as the `currentValue` and match on the `currentDigest`.

**Usage example**

The following is an example where you would maintain the HEAD digest of the `main` branch of a repository.
You would configure a generic regex manager in `renovate.json` for files named `versions.ini`:

```json
{
  "regexManagers": [
    {
      "fileMatch": ["^versions.ini$"],
      "matchStrings": ["GOOGLE_API_VERSION=(?<currentDigest>.*?)\\n"],

>       "currentValueTemplate": "main",

Reviewer 2: I don't think it matters too much, but ideally it should be correct at time of commit, regardless of main or master
